packages/web-components/src/components/cta-section/cta-section.ts

```diff
@@ -35,20 +37,51 @@ const slotExistencePropertyNames = {
  */
 @customElement(`${ddsPrefix}-cta-section`)
 class DDSCTASection extends StableSelectorMixin(DDSContentItem) {
+  /**
+   * Content Items wrapper.
+   */
+  @query(`.${prefix}--content-item-wrapper`)
+  private _contentsNode?: HTMLElement;
+
+  /**
+   * Content Item slot node
+   */
+  @query(`slot[name='content-item']`)
+  private _slotNode?: HTMLSlotElement;
+
   /**
    * `true` if there is CTA content.
    */
   @internalProperty()
   protected _hasFooter = false;
 
   /**
-   * Applies section attribute
+   * The observer for the resize of the content item slot
    */
-  connectedCallback() {
-    if (!this.hasAttribute('role')) {
-      this.setAttribute('role', 'section');
+  private _observerResizeContainer: any | null = null; // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+
+  /**
+   * Cleans-up and creats the resize observer for cta-section
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new resize observer.
+   */
+  private _cleanAndCreateObserverResize({ create }: { create?: boolean } = {}) {
+    const { _contentsNode: contentsNode } = this;
+
+    if (contentsNode?.children[0]?.assignedNodes().length) {
```

Comment:
```suggestion
    const { _slotNode: slotNode } = this;

    if (slotNode?.assignedNodes().length) {
```

Also, given we are checking if the `<slot>` has any assigned nodes, seems that we need to run this method upon `slotchange` event.

Reply:
I see, seems that we won't need `_contentsNode` anymore, but could you explain why we changed the query to [this](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/4511#discussion_r526504287) earlier? Given that the wrapper is a parent element of the actual slot element we're focusing on, it seemed slightly redundant to target it over the child slot itself no?

Reply:
@IgnacioBecerra Good question, it was for the container `ResizeObserver` observes for its size. Seems that you changed it to `this`. So my question is; Does `contentsNode` resize?

Reply:
Hmm, `contentsNode` resizes when the window also does, which I believe also affects the child elements in this case. 

I think I understand -- we were only watching `contentsNode` for resize before, but now that `ResizeObserver` observes `this` (the greater component itself), once the window resize, all the elements within also resize, so there's no need for `contentsNode` now that it's not being observed directly., I see.

Reply:
Thank you for explaining @IgnacioBecerra! If `contentsNode` resizes, we can observe the size of `contentsNode` directly. e.g. What if somebody creates a draggable splitter UI and put our CTA section in either side of the split pane? (Where CTA section resizes without window resizing)

Reply:
Ohh, I see, I get it now. I did not write the rest of your earlier suggestion of using `observe(this._contentsNode)` and left it at `observe(this)` instead. Even though using `this` still worked, having that realization + your splitter UI example made everything click. Thanks @asudoh!